app/src/main/java/net/gsantner/markor/format/general/CommonTextActions.java

```diff
@@ -134,6 +138,20 @@ public boolean runAction(String action) {
                 });
                 return true;
             }
+            case ACTION_MOVE_UP: {
+                moveLineBy1(true);
+                return true;
+            }
+            case ACTION_MOVE_DOWN: {
+                moveLineBy1(false);
+                return true;
+            }
+            case ACTION_NEW_LINE_BELOW: {
+                // Go to end of line, works with wrapped lines too
+                _hlEditor.setSelection(StringUtils.getLineEnd(_hlEditor.getText(), StringUtils.getSelection(_hlEditor)[1]));
```

Comment:
safe regarding Index out of bounds?

Reply:
Both `getLineEnd` and `setSelection` have defensive checks. I've been using this in my own branch for a long time now. Should be ok.